File: src/main/java/me/lokka30/levelledmobs/misc/StringReplacer.java
package me.lokka30.levelledmobs.misc;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.function.Supplier;

public class StringReplacer {
    public StringReplacer(final @NotNull String text){
        this.text = text;
    }

    public @NotNull String text;

    public void replaceIfExists(final @NotNull String target, final @Nullable Supplier<String> operation){
        final String newText = operation != null ? operation.get() : null;

        if (text.contains(target)){
            replace(target, newText);
        }
    }

    public void replace(final @NotNull String replace, final double replaceWith){
        replace(replace, String.valueOf(replaceWith));
    }

    public void replace(final @NotNull String replace, final int replaceWith){
        replace(replace, String.valueOf(replaceWith));
    }

    public StringReplacer replace(final @NotNull String replace, final @Nullable String replaceWith){
        final String replaceWithText = replaceWith != null ? replaceWith : "";

        text = text.replace(replace, replaceWithText);
        return this;
    }

    public boolean isEmpty(){
        return this.text.isEmpty();
    }

    public boolean contains(final @NotNull CharSequence s){
        return this.text.contains(s);
    }
}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/misc/StringReplacer.kt
package io.github.arcaneplugins.levelledmobs.misc

import java.util.function.Supplier

/**
 * Holds a string and provides methods to run
 * various string replacement operations
 *
 * @author stumper66
 * @since 3.13.2
 */
class StringReplacer(var text: String) {
    fun replaceIfExists(target: String, operation: Supplier<String?>?) {
        val newText = operation?.get()

        if (text.contains(target)) {
            replace(target, newText)
        }
    }

    fun replace(replace: String, replaceWith: Double) {
        replace(replace, replaceWith.toString())
    }

    fun replace(replace: String, replaceWith: Int) {
        replace(replace, replaceWith.toString())
    }

    fun replace(replace: String, replaceWith: String?): StringReplacer {
        val replaceWithText = replaceWith ?: ""

        text = text.replace(replace, replaceWithText)
        return this
    }

    val isEmpty: Boolean
        get() = text.isEmpty()


    fun contains(s: CharSequence): Boolean {
        return text.contains(s)
    }
}